src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -1874,17 +1883,14 @@ private void getCallInfo(ref CORINFO_RESOLVED_TOKEN pResolvedToken, CORINFO_RESO
             }
             else
             {
-                if (targetMethod.IsConstructor && targetMethod.OwningType.IsString)
-                {
-                    // Calling a string constructor doesn't call the actual constructor.
-                    targetMethod = targetMethod.GetStringInitializer();
-                }
-
+                pResult.nullInstanceCheck = true;
+                //pResult.kind = CORINFO_CALL_KIND.CORINFO_VIRTUALCALL_STUB;
                 pResult.kind = CORINFO_CALL_KIND.CORINFO_CALL;
```

Comment:
This should be CORINFO_CALL_CODE_POINTER. CORINFO_CALL tells the JIT that it is a direct call and JIT tries to inline it, etc. - that's not the right behavior for interfaces.